src/brsTypes/components/RoSGNode.ts

```diff
@@ -1073,47 +1071,76 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
             return new BrsString(this.type);
         },
     });
+
+    /**
+     * Takes a list of models and creates fields with default values, and adds them to this.fields.
+     * By default, it will not override existing fields.
+     */
+    protected registerFields(fields: FieldModel[], allowOverrides: boolean = false) {
+        fields.forEach(field => {
+            let fieldName = field.name.toLowerCase();
+            if (!allowOverrides && this.fields.has(fieldName)) {
```

Comment:
This is so that we don't overwrite our instantiated `members` with default field values.

Reply:
Can you clarify what this is needed for?  If _registerFields_ is only used for _brs_-provided components is this to prevent a maintainer from accidentally adding a field in a subcomponent that overwrites one in the parent?  Should this silently raise an error?  Should this fail on the first field?  That seems hard to debug.  When will we ever allow overrides?  

If this will be used by custom components is this the desired behavior?  What does the Roku do in this situation?  
```
<component name="A">
    <interface>
        <field name="foo" type="integer">
    </interface>
</component>

<component name="B" extends="A">
    <interface>
        <field name="foo" type="string">
    </interface>
</component>

```

Reply:
Great questions! You bring up a good point about `interface`s. The problem I was trying to solve was this:
```
<Group opacity="0.5" />
```
In this scenario, we would first add `members` to `this.fields` in the `RoSGNode` constructor. Then, we would go through the _default_ fields (and `opacity` is a default field), so the custom `opacity` value that we got from the XML would be overwritten by the default.

I think it might be more clear to have a `registerDefaultFields` and `registerMembers`, and have the child call those methods in the appropriate order, i.e.:

```
class Group {
  constructor(members, name) {
    super([], name); // note that we will not pass members into the super constructor

    this.registerDefaultFields(this.defaultFields);
    this.registerMembers(members);
  }
}
```

Reply:
Do you see any case where we'd want to override members with defaults? I can't think on any right now. If you think `allowOverrides` will never be used I'd suggest to go for the second approach.

Reply:
Yep agreed, I just made those changes!